tests/pytests/functional/transport/rabbitmq/test_async_pub_channel.py

```diff
@@ -0,0 +1,229 @@
+    def _handle_signals(self, signum, sigframe):
+        self.close()
+        super()._handle_signals(signum, sigframe)
+
+    def close(self):
+        if self._closing:
+            return
+        self._closing = True
+        if self.process_manager is None:
+            return
+        self.process_manager.stop_restarting()
+        self.process_manager.send_signal_to_processes(signal.SIGTERM)
+        self.process_manager.kill_children()
+        # Really terminate any process still left behind
+        for pid in self.process_manager._process_map:
+            terminate_process(pid=pid, kill_children=True, slow_stop=False)
+        self.process_manager = None
+
+    def publish(self, payload):
+        self.queue.put(payload)
+
+    def __enter__(self):
+        self.start()
+        self.collector.__enter__()
+        attempts = 30
+        while attempts > 0:
+            self.publish({"tgt_type": "glob", "tgt": "*", "jid": -1, "start": True})
+            if self.collector.running.wait(1) is True:
+                break
+            attempts -= 1
+        else:
+            pytest.fail("Failed to confirm the collector has started")
+        return self
+
+    def __exit__(self, *args):
+        # Publish a payload to tell the collection it's done processing
+        self.publish({"tgt_type": "glob", "tgt": "*", "jid": -1, "stop": True})
+        # Now trigger the collector to also exit
+        self.collector.__exit__(*args)
+        # We can safely wait here without a timeout because the Collector instance has a
+        # hard timeout set, so eventually Collector.stopped will be set
+        self.collector.stopped.wait()
+        # Stop our own processing
+        self.queue.put(None)
+        # Wait at most 10 secs for the above `None` in the queue to be processed
+        self.stopped.wait(10)
+        self.close()
+        self.terminate()
+        log.info("The PubServerChannelProcess has terminated")
+
+
+@pytest.mark.slow_test
+@pytest.mark.xfail(reason="RMQ is POC. Skip RMQ tests until RMQ dependencies are dealt with in the CI/CD pipeline")
+def test_publish_to_pubserv_ipc_async_collector(salt_master, salt_minion):
+    """
+    Test sending a message to RabbitMQPubServerChannel using IPC transport
+    To start rmq container: "docker run  --name rabbitmq -p 5672:5672 -p 15672:15672 bitnami/rabbitmq"
+    """
+    opts = dict(salt_master.config.copy(), ipc_mode="ipc", pub_hwm=0)
+    # this starts the receiver that blocks
+    with PubServerChannelProcess(opts, salt_minion.config.copy()) as server_channel:
+        send_num = 4
+        expect = []
+        for idx in range(send_num):
+            expect.append(idx)
+            load = {"tgt_type": "glob", "tgt": "*", "jid": idx}
+            server_channel.publish(load) # publish N messages
+    results = server_channel.collector.results
+    assert len(results) == send_num, "{} != {}, difference: {}".format(
+        len(results), send_num, set(expect).difference(results)
```

Comment:
**question** is the `results` just a list of numbers?

Reply:
Yes, the payload is just a bunch sequential numbers.